api/controller.py

```diff
@@ -608,6 +608,12 @@ def handle_conditional_request(self, last_modified=None):
         if not last_modified:
             return None
 
+        # If-Modified-Since values have resolution of one second. If
+        # last_modified has millisecond resolution, change its
+        # resolution to one second.
+        if last_modified.microsecond:
+            last_modified = last_modified.replace(microsecond=0)
+
```

Comment:
It seems like this would be better captured in the `last_loan_activity_sync` getter and setter in the `patron` model.

Reply:
I think this is more an issue of what's possible with the time format used by HTTP. If I could put microsecond data in the Last-Modified header, I would. If we use last_loan_activity_sync for other purposes, then we might want microsecond resolution there. If the timeout is set to a very short period, the microsecond data will become useful when doing time comparisons between last_loan_activity_sync and the current time--half a second one way or another will make a relatively large difference.

That said, if we use last_loan_activity_sync for other purposes and this keeps being a problem, I'd be OK with moving the resolution change into the model.